packages/ai-jsx/src/lib/cohere.tsx

```diff
@@ -0,0 +1,145 @@
+import * as AI from '../index.js';
+import { getEnvVar } from './util.js';
+
+/**
+ * Properties for a {@link cohereContext} that configure the Cohere API.
+ */
+export interface CohereProps {
+  api_url?: string;
+  api_key?: string;
+
+  /**
+   * Model to be used for reranking.
+   *
+   * @see https://docs.cohere.com/docs/reranking#parameters
+   */
+  reranker_model_name?: string;
+}
+
+export const DEFAULT_COHERE_CONFIGURATION = {
+  reranker_model_name: 'rerank-multilingual-v2.0',
+  api_url: 'https://api.cohere.ai/v1',
+  api_key: getEnvVar('COHERE_API_KEY', false),
+};
+
+/**
+ * Allows for configuring Cohere API
+ * Uses COHERE_API_KEY environment variable and the multilingual model by default
+ *
+ * Currently only used by {@link reranker}.
+ */
+export const cohereContext = AI.createContext<CohereProps>(DEFAULT_COHERE_CONFIGURATION);
+
+/**
+ * Reranker from Cohere
+ * Givne a query and a list of documents, returns the top_n documents sorted by relevance
+ *
+ * @see https://docs.cohere.com/docs/reranking
+ *
+ * @returns An array of objects with the index of the document and its relevance score
+ */
+export async function reranker(
+  {
+    query,
+    documents,
+    top_n,
+    model,
+  }: {
+    query: string;
+    documents: string[];
+    top_n: number;
+    model?: string;
+  },
+  { getContext, logger }: Pick<AI.ComponentContext, 'getContext' | 'logger'>
+): Promise<{ index: number; relevance_score: number }[]> {
+  const cohereConfig = getContext(cohereContext);
+  if (!cohereConfig.api_key) {
+    logger.error(
+      { CohereContext: cohereConfig },
+      'Cohere API key is not set, but it is needed for doing reranking (e.g. loadBySimilarity).' +
+        ' Please set it in the context.'
+    );
+    // fallback to original order
```

Comment:
```suggestion
    // fall back to original order
```

"fallback" is a noun; "fall back" is a verb